flink-runtime/src/main/java/org/apache/flink/runtime/memory/AbstractPagedInputView.java

```diff
@@ -398,9 +398,7 @@ public String readLine() throws IOException {
 		try {
 			int b;
 			while ((b = readUnsignedByte()) != '\n') {
-				if (b != '\r') {
-					bld.append((char) b);
-				}
+				bld.append((char) b);
```

Comment:
I don't think this can be changed, here we remove all the '\r' and line 412 just remove the last '\r'.

Reply:
> I don't think this can be changed, here we remove all the '\r' and line 412 just remove the last '\r'.

Thank you so much for your comment. 
1. If we have removed all the '\r', why do we have to remove the last '\r'?
2. If the '\r' appears in the middle of a line, it is not used as a line separator, and we should not remove it, although this rarely happens in practice.